### ISSUE TITLE
📎 Implement `noUnusedStyles`

### ISSUE THREAD

**Issue by ematipico** [biomejs/biome#9156]
### Description

This rule will detect unused CSS classes across a project e.g. 

```css
/* styles.css */
.foo { color: red }
```

```jsx
// App.jsx
import "styles.css"

function App() {
  return <div>Opsie</div>
}
```

Will result in 

```
styles.css:1:1 lint/nursery/noUnusedStyles ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  i The CSS class foo is not referenced in any HTML or JSX file.
  
  > 1 │ foo { color: red }
      │ ^^^
  
  i Unused CSS classes add dead weight to your stylesheet, increase bundle size, and make the codebase harder to maintain.
  
  i Either use this class somewhere or remove the selector.
```

**Comment by dyc3** [comment]
1. Should this have an `ignore` or `ignoreClasses` option? Sometimes people dynamicly build css classes, or certain styles might come from an external style sheet that is not checked in to source control. If so, should we use globs?
2. Should we consider id selectors? Might be a source of false positives, but IMO I don't think it would be that bad.
3. Should we consider attribute selectors? Might be a source of false positives.
